pkg/bisect/bisect.go

```diff
@@ -302,6 +310,35 @@ func (env *env) bisect() (*Result, error) {
 	return res, nil
 }
 
+func (env *env) identifyRewrittenCommit() (string, error) {
+	cfg := env.cfg
+	_, err := env.repo.CheckoutBranch(cfg.Kernel.Repo, cfg.Kernel.Branch)
+	if err != nil {
+		return cfg.Kernel.Commit, err
+	}
+	contained, err := env.repo.Contains(cfg.Kernel.Commit)
+	if err != nil {
+		return cfg.Kernel.Commit, err
+	}
+	if contained {
+		return cfg.Kernel.Commit, err
+	}
+
+	// We record the tested kernel commit when syzkaller triggers a crash. These commits can become
+	// unreachable after the crash was found, when the history of the tested kernel branch was
+	// rewritten. The commit might have been completely deleted from the branch or just changed in
+	// some way. Some branches like linux-next are often and heavily rewritten (aka rebased).
+	// This can also happen when changing the branch you fuzz in an existing syz-manager config.
+	// This makes sense when a downstream kernel fork rebased on top of a new upstream version and
+	// you don't want syzkaller to report all your old bugs again.
+	commit, err := env.repo.GetCommitByTitle(cfg.Kernel.CommitTitle)
+	if err != nil {
```

Comment:
I am thinking what happens if the old commit is not reachable from HEAD, but the commit title is also not present in the current HEAD. E.g. a linux-next commit was renamed or temporary dropped.
I think at least for cause bisection it still makes sense to bisect using the old commit. What do you think?

Reply:
I think there are two reasons against doing this:
1) It makes it even harder to reason about this code. It's already somewhat magic.
2) We currently don't re-run cause bisection, once a bisection was successful, unless the current bug is marked closed and the same crash is discovered again. Hence for now it is preferable to not bisect a commit that was temporarily dropped. If a new version of the patch with the same bug lands in linux-next again, syzkaller will just find and add a new crash with the new hash/title to the existing bug, causing a fresh cause bisection anyhow.

Does this make sense?

Reply:
Yes.

Reply:
Now (before and after this PR) we are just failing the bisection process in the case the commit was completely dropped, right?

Reply:
Correct (for when the commit is no longer referenced in any way in the repo and was garbage collected by git already).